onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -5609,13 +5613,14 @@ def _aten_native_batch_norm_training_onnx(
         training_mode=training,
     )
     # Compute var and rstd
-    mean = op.ReduceMean(input, axes)
-    input_sub_mean = op.Sub(input, mean)
+    upcast_input = op.Cast(input, to=FLOAT.dtype)
```

Comment:
Could you add a comment to explain why upcasting is needed (I assume it is because of training?) Thanks!